**Name: Maria Master**

**Class: INST 314 (0202)**

## Dataset 2: Breast Cancer Wisconsin (Diagnostic) Data

**1. Describe how you found the data.  If you used a search engine, provide the exact search engine and search terms you used. Otherwise, describe how you came to find the data.**

**A:** I found this dataset in Kaggle after I looked up "Cancer" in the search bar.

**2. Give a short verbal description of the data. Just a couple of sentences, like you were describing it briefly to someone in person, just to tell them generally what it is.**

**A:** This dataset is a collection of digitized images of fine needle aspirate (FNA) of cancerous masses and describes certain characteristics of the cell nuclei present in the image. The variables include measurements such as radius, perimeter, smoothness, etc. 

**3. Do you have any information about how the data was collected?  If so, briefly describe the data collection process.  If not, take a guess how it might have been collected, and describe why you think that.**

**A:** The dataset was collected from digitized images of breast masses that were obtained from fine needle aspirate (FNA) biopsies. "Interactive image processing techniques, along with a linear-programming-based inductive classifier, have been used" as well (Smith, Jones, & Brown, 2020). The data from the images was then digitized. 

Refernce:

Street, W. N., Wolberg, W. H., & Mangasarian, O. L. (1993, July 29). Nuclear feature extraction for breast tumor diagnosis. *Biomedical Image Processing and Biomedical Visualization* (Proc. SPIE 1905). https://doi.org/10.1117/12.148698 

**4. a) Where is the data located (URL, etc.)?  This should be specific enough that the instructor/graders can find and download thedata themselves.If the instructor/graders cannot access or download the data themselves, describe what is preventing that (e.g., this is personal data on your own computer), and describe how you plan to share the data.**

**A**: URL: https://www.kaggle.com/datasets/uciml/breast-cancer-wisconsin-data

If the data can not be accessed using the URL above, I will share the csv file with the instructional team. PLEASE make sure to upload the data in the same workspace as this Jupyter Notebook.

**4. b) File size (in KB, MB, or GB)**

**A:** 125.2 kB

**4. c) File format (extension, CSV, JSON, Excel, etc.)**

**A:** CSV

**5. How do you know you have permission to use and share the data?  For example, if there are any licenses, describe those, or if there is a URL to a page that describes permissions, you can link to that. Otherwise, describe your rationale for why you think you can analyze and share the data.**

**A:** The license is: CC BY-NC-SA 4.0 (Creative Commons Attribute 4.0 International (CC BY 4.0)). It allows for the sharing and adaptation of the dataset as long as UCL Machine Learning Repository is credited.

**6. Do you think there is any kind of privacy or ethical risk by using/sharing the data?  Describe your rationale.**

**A:** I do not think there is any significant issue of privacy or ethical risk when using/sharing this data. The data, which is anonymized, was originally used for a research paper. Therefore, unless there is a way to identify any patients with the data provided, there should not be problem using this data. Lastly, this data should not be used to cause harm in no way.

**7. Load the data into your analysis program (R or Python).  Paste in the code you used that worked.**

In [15]:
# import pandas
import pandas as pd

# create dataframe
df = pd.read_csv("Breast_cancer_wisconsin.csv", sep = ",")

**8. Use R/Python to look at the first few rows of data.  Paste in the code you used to do that.**

In [16]:
# print the top 5 records
print(df.head(5))

         id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  texture_worst  perimeter_worst  area_worst  smoothness

**9. Are you surprised by anything in the first few rows?  What kind of values did you expect to see?  Your answer can be general; you don't have to describe every variable/column.**

**A:** Looking at the first few rows, I am pretty confused with how small some of the numbers are such as the mean of radius, texture, smoothness etc, compared to the big numbers such as the area mean, etc. Lastly, it looks like all the variables are numerical except the diagnosis.

**10. How many columns/variables of numeric data are there?  Note that your data *must* have at least
two columns of numeric data.  Ifnot, keep looking!**

**A:** Other than the Id, there are 30 numerical variables.

**11.  Is there non-numeric data?  What kinds of non-numeric data do there appear to be?**

**A:** Yes, there is 1 non-numeric variable. It is a boolean datatype (in this case, it is either M for malignant or B for benign; M is cancerous and B is non-cancerous).

**12. How many rows/records are there?  How many columns/variables?**

**A:** There are 569 rows/records. There are 32 columns/variables.

**13. Select a numeric column that appears to be interesting.  Describe why you think it might be interesting.**

**A:** An interesting numeric column is "radius_worst". It shows us the largest mean value for the mean of distances from center to points on the perimeter of the mass. The reason why I initially think it is interesting is because I wonder if there is a correlation between a mass being malignant and having a large radius mean. 

**14. Make an educated guess about what you think the minimum and maximum values of this numeric variable are. Describe your reasoning.**

**A:** Judging from the first couple of records, I think the maximum value is around 27 and the minimum is around 8. The radius mean in this case is the mean of the distances from the center of a cell nucleus to the points on its perimeter. Therefore, after taking a look at the first couple of records from the normal radius_mean column, which has a minimum of around 6 and a maximum of around 19, I assumed that the minimum for the worst radius mean would be higher, around 8, and the maximum for the worst radius mean would be higher as well, around 27. 

**15. Get the actual minimum and maximum values, and paste in the code you used to get these values.**

In [17]:
# find the min and max value of the numeric variable "radius_worst"
min_val = df["radius_worst"].min()
max_val = df["radius_worst"].max()

# print the min and max value 
print("Minimum value: ", min_val)
print("Maximum value: ", max_val)

Minimum value:  7.93
Maximum value:  36.04


**16. Were you surprised by the actual values?  Describe what you think about why they are defying your expectations.**

**A:** I was surprised by the actual maximum value. I assumed it to be 27. However, I understand that it was just an assumption I made after going through the first couple of records. Furthermore, it also could mean that the cell nuclei was not normal for all of them. Lastly, the worst radius is measured by taking the mean of the largest values from the normal radius mean column. Therefore, the normal radius mean column is more susceptible to outliers and thus, this could have influenced the worst radius mean in making the final maximum value to be 36.04. 

**17. Can you see any patterns of missing data in this data set? Describe what you see, and paste in code you used to look.**

**A:** There is no missing data. 

In [18]:
""" 
Option 1: 
- Use .isna() pandas function to check whether there is a missing value or not.
- Will return true if yes, and False if no. """

print(df.isna().head(5))

      id  diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0  False      False        False         False           False      False   
1  False      False        False         False           False      False   
2  False      False        False         False           False      False   
3  False      False        False         False           False      False   
4  False      False        False         False           False      False   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0            False             False           False                False   
1            False             False           False                False   
2            False             False           False                False   
3            False             False           False                False   
4            False             False           False                False   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0 

In [19]:
""" 
Option 2:
- Use the .isna() function with .sum() function. This will count any instances of True that 
the .isna() function will return. If the sum is 0, there are no missing record instances."""

print(df.isna().sum())

id                           0
diagnosis                    0
radius_mean                  0
texture_mean                 0
perimeter_mean               0
area_mean                    0
smoothness_mean              0
compactness_mean             0
concavity_mean               0
concave points_mean          0
symmetry_mean                0
fractal_dimension_mean       0
radius_se                    0
texture_se                   0
perimeter_se                 0
area_se                      0
smoothness_se                0
compactness_se               0
concavity_se                 0
concave points_se            0
symmetry_se                  0
fractal_dimension_se         0
radius_worst                 0
texture_worst                0
perimeter_worst              0
area_worst                   0
smoothness_worst             0
compactness_worst            0
concavity_worst              0
concave points_worst         0
symmetry_worst               0
fractal_dimension_worst      0
Unnamed:

**18. Describe anything else you found surprising or noteworthy about the data.**

**A:** It seems like the texture and size measurements of a cell nuclei are important factors in determining cancerous masses when it comes to breast cancer. Other characteristics such as compactness and smoothness could aso possibly be important, but it seems as though it might not be as crucial as the texture and size measurements of the cell nuclei (this can only be confirmed after analyzing the data).

**19. Is this data interesting to you?  Why or why not?  What kinds of questions do you think it might be able to answer?**

**A:** This dataset is interesting to me. After analyzing this data, it might give us valuable insights on inventing newer methods to pay close attention to leading factors that can be an early indication of a malignant tumor. Some questions that can be answered are:

- Are the size and texture measurements of the cell nuclei leading factors of malignant tumors?
- Does other features of the mass which includes the smoothness, concavity, compactness, have any association with the cell nuclei measurements?
- Can this dataset be used to potentially come up with better machine learning models to predict early signs of breast cancer?

These are only some of the questions. There could possibly be many more.